src/main/java/it/pagopa/pn/deliverypush/legalfacts/PdfboxLegalFactPdfGenerator.java

```diff
@@ -24,338 +23,397 @@
-	private static final String PARAGRAPH1 = "Ai sensi dell’art. 26, comma 11, del decreto-legge,"
-			+ " la PagoPA s.p.a. nella sua qualità di gestore ex lege"
-			+ " ella Piattaforma Notifiche Digitali di cui allo stesso art. 26,"
-			+ " con ogni valore legale per l'opponibilità a terzi, ATTESTA CHE:";
-	
-	private static final PDFont FONT = PDType1Font.TIMES_ROMAN;
+    private static final String PARAGRAPH1 = "Ai sensi dell’art. 26, comma 11, del decreto-legge,"
+            + " la PagoPA s.p.a. nella sua qualità di gestore ex lege"
+            + " ella Piattaforma Notifiche Digitali di cui allo stesso art. 26,"
+            + " con ogni valore legale per l'opponibilità a terzi, ATTESTA CHE:";
+
+    private static final PDFont FONT = PDType1Font.TIMES_ROMAN;
     private static final float FONT_SIZE = 12;
     private static final float LEADING = -1.5f * FONT_SIZE;
     private static final float TY_NEW_PAGE = 186f;
-    
+
     private static final float MARGINY = 120f;
     private static final float MARGINX = 80f;
-    
+
     private PDDocument document = null;
     private PDPageContentStream contentStream = null;
     private float calculatedTY;
-    
+
     private float startX;
     private float startY;
+
+    @Autowired
+    public PdfboxLegalFactPdfGenerator(TimelineDao timelineDao) {
+        super(timelineDao);
+    }
+
+    @Override
+    public byte[] generateNotificationReceivedLegalFact(Action action, Notification notification) {
+        String paragraph2 = "in data %s il soggetto mittente %s, C.F. "
+                + "%s ha messo a disposizione del gestore i documenti informatici di "
+                + "cui allo IUN %s e identificati in modo univoco con i seguenti hash: ";
+        paragraph2 = String.format( paragraph2, this.instantToDate( notification.getSentAt() ),
+                notification.getSender().getPaDenomination(),
+                notification.getSender().getTaxId( notification.getSender().getPaId() ),
+                action.getIun());
+        StringBuilder bld = new StringBuilder();
+        for (int idx = 0; idx < notification.getDocuments().size(); idx ++) {
+            bld.append( "\n" + notification.getDocuments().get(idx).getDigests().getSha256() + "; " );
+        }
+
+        if ( notification.getPayment() != null && notification.getPayment().getF24() != null ) {
+            if ( notification.getPayment().getF24().getFlatRate() != null) {
+                bld.append( "\n" + notification.getPayment().getF24().getFlatRate().getDigests().getSha256() + ";" );
+            }
+            if ( notification.getPayment().getF24().getDigital() != null) {
+                bld.append( "\n" + notification.getPayment().getF24().getDigital().getDigests().getSha256() + ";" );
+            }
+            if ( notification.getPayment().getF24().getAnalog() != null) {
+                bld.append( "\n" + notification.getPayment().getF24().getAnalog().getDigests().getSha256() + ";" );
+            }
+        }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 79:89, 131:141.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.

Reply:
Possiamo ingnorare uesta segnalazione in quanto la classe è da rifattorizzare per intero